### PR TITLE
Add eval.py and fix bug for MobileNet-SSD.

### PR DESCRIPTION
**Pull request by qingqing01**
Fix https://github.com/PaddlePaddle/models/issues/853

### CODE REVIEW

File: fluid/object_detection/eval.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+add_arg('test_list',        str,   '',        "The path to save model.")
+add_arg('label_file',       str,   '',        "Label file.")
+add_arg('model_dir',        str,   '',        "The path to save model.")
+add_arg('ap_version',       str,  '11point',  "11point or integral")
+add_arg('resize_h',         int,  300,    "resize image size")
+add_arg('resize_w',         int,  300,    "resize image size")
+add_arg('mean_value_B',     float, 127.5, "mean value which will be subtracted")  #123.68
+add_arg('mean_value_G',     float, 127.5, "mean value which will be subtracted")  #116.78
+add_arg('mean_value_R',     float, 127.5, "mean value which will be subtracted")  #103.94
+# yapf: disable
```
Comment (wanghaoshuang) [inline fluid/object_detection/eval.py]: 这里好像应该用`yapf: enable` https://github.com/google/yapf#why-does-yapf-destroy-my-awesome-formatting

File: fluid/object_detection/eval.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+from mobilenet_ssd import mobile_net
+from utility import add_arguments, print_arguments
+
+parser = argparse.ArgumentParser(description=__doc__)
+add_arg = functools.partial(add_arguments, argparser=parser)
+# yapf: disable
+add_arg('dataset',          str, 'pascalvoc', "coco or pascalvoc.")
+add_arg('batch_size',       int,   32,        "Minibatch size.")
+add_arg('use_gpu',          bool,  True,      "Whether use GPU.")
+add_arg('data_dir',         str,   '',        "The path to save model.")
```
Comment (wanghaoshuang) [inline fluid/object_detection/eval.py]: ~~The path to save model.~~ Please update the comments.

File: fluid/object_detection/eval.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+add_arg('test_list',        str,   '',        "The path to save model.")
```
Comment (wanghaoshuang) [inline fluid/object_detection/eval.py]: Please update the comments.

File: fluid/object_detection/eval.py (hunk 1 above)
Comment (qingqing01) [reply fluid/object_detection/eval.py]: Done.

File: fluid/object_detection/eval.py (hunk 2 above)
Comment (qingqing01) [reply fluid/object_detection/eval.py]: Done.

File: fluid/object_detection/eval.py (hunk 3 above)
Comment (qingqing01) [reply fluid/object_detection/eval.py]: Done.

File: fluid/object_detection/eval.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            gt_box,
+            difficult,
+            num_classes,
+            overlap_threshold=0.5,
+            evaluate_difficult=False,
+            ap_version=args.ap_version)
+
+    place = fluid.CUDAPlace(0) if args.use_gpu else fluid.CPUPlace()
+    exe = fluid.Executor(place)
+    #exe.run(fluid.default_startup_program())
```
Comment (wanghaoshuang) [inline fluid/object_detection/eval.py]: Remove this line?

File: fluid/object_detection/eval.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+    if model_dir:
+
+        def if_exist(var):
+            return os.path.exists(os.path.join(model_dir, var.name))
+
+        fluid.io.load_vars(exe, model_dir, predicate=if_exist)
+    #fluid.io.load_persistables(exe, model_dir, main_program=test_program)
```
Comment (wanghaoshuang) [inline fluid/object_detection/eval.py]: Remove this line?

File: fluid/object_detection/train.py [hunk 6]
```diff
@@ -234,36 +249,48 @@ def if_exist(var):
...
+        print 'save models to %s' % (model_path)
+        fluid.io.save_persistables(exe, model_path)
+
+    best_map = 0.
+
+    def test(pass_id, best_map):
         _, accum_map = map_eval.get_map_var()
         map_eval.reset(exe)
         test_map = None
         for _, data in enumerate(test_reader()):
```
Comment (wanghaoshuang) [inline fluid/object_detection/train.py]: 这里为什么要用enumerate呢？

File: fluid/object_detection/reader.py [hunk 7]
```diff
@@ -98,15 +102,95 @@ def img_mean(self):
...
+                sampled_labels[i][3] = 1 - tmp
+    # HWC to CHW
+    if len(img.shape) == 3:
+        img = np.swapaxes(img, 1, 2)
+        img = np.swapaxes(img, 1, 0)
+    # RBG to BGR
+    img = img[[2, 1, 0], :, :]
+    img = img.astype('float32')
+    img -= settings.img_mean
+    #img = img.flatten()
```
Comment (wanghaoshuang) [inline fluid/object_detection/reader.py]: 这行还需要么？

File: fluid/object_detection/eval.py (hunk 5 above)
Comment (qingqing01) [reply fluid/object_detection/eval.py]: Done. Thanks!

File: fluid/object_detection/reader.py (hunk 7 above)
Comment (qingqing01) [reply fluid/object_detection/reader.py]: Removed. Thanks!

File: fluid/object_detection/train.py (hunk 6 above)
Comment (qingqing01) [reply fluid/object_detection/train.py]: Done. Thanks!

File: fluid/object_detection/eval.py (hunk 4 above)
Comment (qingqing01) [reply fluid/object_detection/eval.py]: Done. Thanks!

**Review by wanghaoshuang — APPROVED** [review]
LGTM.

**Pull request merged**